src/transformState.js

```diff
@@ -5,7 +5,24 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(state, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keys of obj.keysToRemove) {
+          delete state[keys];
+        }
+        break;
+
+      default:
+        for (const key in state) {
+          delete state[key];
+        }
```

Comment:
This logic should be moved to a separate `case`.

In the `default` block you should handle errors